### ISSUE TITLE
prefer-array-flat-map false positives for React.children

### ISSUE THREAD

**Issue by skovhus** [oxc-project/oxc#13531]
Latest oxlint. When applying auto-fix for `prefer-array-flat-map` introduces false positives for React.Children.

```jsx
          React.Children.map(props.children, child => {
            if (typeof child === "string") {
              return child.split("+");
            }

            return "";
          })
            ?.flat()
```

ends up as
```jsx
          React.Children.flatMap(props.children, child => {
            if (typeof child === "string") {
              return child.split("+");
            }

            return "";
          })
```

Note that `React.Children` is not an array.
